src/runtime/pkg/kata-monitor/monitor.go

```diff
@@ -29,55 +31,47 @@ func SetLogger(logger *logrus.Entry) {
 
 // KataMonitor is monitor agent
 type KataMonitor struct {
-	sandboxCache         *sandboxCache
-	containerdAddr       string
-	containerdConfigFile string
-	containerdStatePath  string
+	runtimeEndpoint string
+	sandboxCache    *sandboxCache
 }
 
 // NewKataMonitor create and return a new KataMonitor instance
-func NewKataMonitor(containerdAddr, containerdConfigFile string) (*KataMonitor, error) {
-	if containerdAddr == "" {
-		return nil, fmt.Errorf("containerd serve address missing")
-	}
-
-	containerdConf := &srvconfig.Config{
-		State: defaults.DefaultStateDir,
+func NewKataMonitor(runtimeEndpoint string) (*KataMonitor, error) {
+	if runtimeEndpoint == "" {
+		return nil, fmt.Errorf("runtime endpoint missing")
 	}
 
-	if err := srvconfig.LoadConfig(containerdConfigFile, containerdConf); err != nil && !os.IsNotExist(err) {
-		return nil, err
+	if !strings.HasPrefix(runtimeEndpoint, "unix") {
+		runtimeEndpoint = "unix://" + runtimeEndpoint
 	}
 
 	km := &KataMonitor{
-		containerdAddr:       containerdAddr,
-		containerdConfigFile: containerdConfigFile,
-		containerdStatePath:  containerdConf.State,
+		runtimeEndpoint: runtimeEndpoint,
 		sandboxCache: &sandboxCache{
 			Mutex:     &sync.Mutex{},
-			sandboxes: make(map[string]string),
+			sandboxes: make(map[string]struct{}),
 		},
 	}
 
-	if err := km.initSandboxCache(); err != nil {
-		return nil, err
-	}
-
 	// register metrics
 	registerMetrics()
 
-	go km.sandboxCache.startEventsListener(km.containerdAddr)
+	go km.startPodCacheUpdater()
 
 	return km, nil
 }
 
-func (km *KataMonitor) initSandboxCache() error {
-	sandboxes, err := km.getSandboxes()
-	if err != nil {
-		return err
+// startPodCacheUpdater will boot a thread to manage sandbox cache
+func (km *KataMonitor) startPodCacheUpdater() {
+	for {
+		sandboxes, err := km.getSandboxes()
+		if err != nil {
+			monitorLog.WithError(err).Error("failed to get sandboxes")
```

Comment:
`continue` when error to skip updating cache. but a `Sleep` may be needed to avoid getting sandboxes too frequently. So how about:

- move `time.Sleep` to the top in the `for` loop
- add `continue` if `err != nil`

Reply:
Yep, done!